warmrestart/warmRestartHelper.cpp

```diff
@@ -274,25 +274,24 @@ void WarmStartHelper::reconcile(void)
 bool WarmStartHelper::compareAllFV(const std::vector<FieldValueTuple> &v1,
                                    const std::vector<FieldValueTuple> &v2)
 {
+    /* Size mismatch implies a diff */
+    if (v1.size() != v2.size())
+    {
+        return true;
+    }
+
     std::unordered_map<std::string, std::string> v1Map((v1.begin()), v1.end());
 
      /* Iterate though all v2 tuples to check if their content match v1 ones */
     for (auto &v2fv : v2)
     {
         auto v1Iter = v1Map.find(v2fv.first);
-        /*
-         * The sizes of both tuple-vectors should always match within any
-         * given application. In other words, all fields within v1 should be
-         * also present in v2.
-         *
-         * To make this possible, every application should continue relying on a
-         * uniform schema to create/generate information. For example, fpmsyncd
-         * will be always expected to push FieldValueTuples with "nexthop" and
-         * "ifname" fields; neighsyncd is expected to make use of "family" and
-         * "neigh" fields, etc. The existing reconciliation logic will rely on
-         * this assumption.
-         */
-        assert(v1Iter != v1Map.end());
+
+        /* Return true when v2 has a new field */
+        if (v1Iter == v1Map.end())
+        {
+            return true;
```

Comment:
should this be continue?

Reply:
how about the fields that are same?

Reply:
In that case, this logic is invoked. And if compareOneFV return false, this would continue with the next field.

```
if (compareOneFV(v1Map[fvField(*v1Iter)], fvValue(v2fv)))
{
    return true;
}
```

Reply:
Not required, we just have to check if there is a diff or not (if v2 has a field that's not present in v1 eg: protocol, it already implies a diff)

If diff is found we push the entire v2 down to orchagent

```
if (compareAllFV(restoredFV, refreshedFV))
{
    SWSS_LOG_NOTICE("Warm-Restart reconciliation: updating entry %s",
                    printKFV(refreshedKey, refreshedFV).c_str());

    m_syncTable->set(refreshedKey, refreshedFV);
}
else
{
    SWSS_LOG_INFO("Warm-Restart reconciliation: no changes needed for "
                  "existing entry %s",
  }
```